src/core/crypto/Utilities.ts

```diff
@@ -201,26 +201,24 @@ export const catapult_crypto = (function() {
             return 0 === c.crypto_verify_32(signature, 0, t, 0);
         },
 
-        deriveSharedKey: (salt, sk, pk, hashfunc) => {
+        deriveSharedKey: (salt, sk, pk, hashfunc, signSchema: SignSchema) => {
             const c = nacl;
-            const d = prepareForScalarMult(sk, hashfunc);
+            const d = prepareForScalarMult(sk, hashfunc, signSchema);
 
             // sharedKey = pack(p = d (derived from sk) * q (derived from pk))
             const q = [c.gf(), c.gf(), c.gf(), c.gf()];
             const p = [c.gf(), c.gf(), c.gf(), c.gf()];
             const sharedKey = new Uint8Array(Key_Size);
-            c.unpackneg(q, pk);
+            c.unpack(q, pk);
```

Comment:
why this change ?

Reply:
`unpack` is used NIS1 code. I noticed without using it, the last byte of the hash is different.

Reply:
what does catapult-server source code say about this?

Reply:
where do I find this 😆 ?